frontend/src/metabase/querying/filters/utils/dates.ts

```diff
@@ -222,22 +246,114 @@ function getQuarterYearFilterClause(
   });
 }
 
-export function getDatePickerOperators(
-  query: Lib.Query,
-  stageIndex: number,
-  column: Lib.ColumnMetadata,
-): DatePickerOperator[] {
-  return Lib.filterableColumnOperators(column)
-    .map(operator => Lib.displayInfo(query, stageIndex, operator).shortName)
-    .filter(isDatePickerOperator);
+export function getDateFilterDisplayName(
+  value: DateFilterValue,
+  { withPrefix }: DateFilterDisplayOpts = {},
+) {
+  return match(value)
+    .with(
+      { type: "specific", operator: "=" },
+      ({ values: [date], hasTime }) => {
+        const dateText = formatDate(date, hasTime);
+        return withPrefix ? t`On ${dateText}` : dateText;
+      },
+    )
+    .with(
+      { type: "specific", operator: "<" },
+      ({ values: [date], hasTime }) => {
+        return t`Before ${formatDate(date, hasTime)}`;
+      },
+    )
+    .with(
+      { type: "specific", operator: ">" },
+      ({ values: [date], hasTime }) => {
+        return t`After ${formatDate(date, hasTime)}`;
+      },
+    )
+    .with(
+      { type: "specific", operator: "between" },
+      ({ values: [startDate, endDate], hasTime }) => {
+        return `${formatDate(startDate, hasTime)} - ${formatDate(endDate, hasTime)}`;
+      },
+    )
+    .with({ type: "relative" }, ({ value, unit, offsetValue, offsetUnit }) => {
+      if (offsetValue != null && offsetUnit != null) {
+        const prefix = Lib.describeTemporalInterval(value, unit);
+        const suffix = Lib.describeRelativeDatetime(offsetValue, offsetUnit);
+        return t`${prefix}, starting ${suffix}`;
+      } else {
+        return Lib.describeTemporalInterval(value, unit);
+      }
+    })
+    .with({ type: "exclude", operator: "!=" }, ({ values, unit }) => {
+      if (values.length <= 2 && unit != null) {
+        const parts = values.map(value => formatExcludeUnit(value, unit));
+        return t`Exclude ${parts.join(", ")}`;
+      } else {
+        const count = values.length;
+        return ngettext(
+          msgid`Exclude ${count} selection`,
+          `Exclude ${count} selections`,
+          count,
+        );
+      }
+    })
+    .with({ type: "exclude", operator: "is-null" }, () => {
+      return t`Is empty`;
+    })
+    .with({ type: "exclude", operator: "not-null" }, () => {
+      return t`Not empty`;
+    })
+    .with({ type: "month" }, ({ month, year }) => {
+      return formatMonth(month, year);
+    })
+    .with({ type: "quarter" }, ({ quarter, year }) => {
+      return formatQuarter(quarter, year);
+    })
+    .exhaustive();
 }
 
-export function getDatePickerUnits(
-  query: Lib.Query,
-  stageIndex: number,
-  column: Lib.ColumnMetadata,
-): DatePickerUnit[] {
-  return Lib.availableTemporalBuckets(query, stageIndex, column)
-    .map(operator => Lib.displayInfo(query, stageIndex, operator).shortName)
-    .filter(isDatePickerUnit);
+function formatDate(date: Date, hasTime: boolean) {
+  return hasTime
+    ? dayjs(date).format("MMMM D, YYYY hh:mm A")
```

Comment:
Should we respect [date/time formatting settings](https://www.metabase.com/docs/latest/configuring-metabase/localization) in these helpers?

Reply:
We didn't before.